src/main/kotlin/edu/carroll/cs389/service/tracker/TrackService.kt

```diff
@@ -0,0 +1,45 @@
+package edu.carroll.cs389.service.tracker
+
+import edu.carroll.cs389.jpa.model.TrackedUser
+import jakarta.servlet.http.HttpServletRequest
+
+/**
+ * Interface which defines the methods required by a TrackService
+ * Indented purpose is to track client connection information when the client accesses any page within the webapp
+ *
+ * Currently, this class tracks connecting ipv4 address, browser & version, operating system & version, the page the request
+ * was directed to, and the timestamp of the request
+ */
+interface TrackService {
+    /**
+     * Method which processes incoming requests and converts them into a storable entity in the form of a TrackedUser
+     * Once a TrackedUser instance is created from the data in the passed servlet request, it is then saved to the
```

Comment:
...it is then persisted. 

You don't have to give the details of what you are using for persistence, as the point of the service is to abstract the details of what technology is persisting the data is not important.